### PR TITLE
[Transform] Add yml test suite for testing remote clusters (CCS)

### PR DESCRIPTION
**Pull request by hendrikmuhs**
add a test suite for remote clusters features and add test cases for transform

### CODE REVIEW

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/2
run elasticsearch-ci/bwc
run elasticsearch-ci/default-distro

**Comment by hendrikmuhs** [comment]
@davidkyle May I ask you to review this, you have a lot of knowledge about yaml tests.

File: x-pack/qa/multi-cluster-tests-with-security/src/test/resources/rest-api-spec/test/multi_cluster/80_transform.yml [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+  - do:
+      headers: { Authorization: "Basic am9lOnRyYW5zZm9ybQ==" }
+      transform.get_transform_stats:
+        transform_id: "simple-remote-transform"
+  - match: { count: 1 }
+  - match: { transforms.0.id: "simple-remote-transform" }
+  - match: { transforms.0.state: "stopped" }
+  - match: { transforms.0.checkpointing.last.checkpoint: 1 }
+
+  # workaround: refresh dest index, to be removed, see gh #TBD
```
Comment (davidkyle) [inline x-pack/qa/multi-cluster-tests-with-security/src/test/resources/rest-api-spec/test/multi_cluster/80_transform.yml]: #TBD -> #51154

File: x-pack/qa/multi-cluster-tests-with-security/src/test/resources/rest-api-spec/test/multi_cluster/80_transform.yml [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+  - do:
+      headers: { Authorization: "Basic am9lOnRyYW5zZm9ybQ==" }
+      transform.get_transform_stats:
+        transform_id: "simple-local-remote-transform"
+  - match: { count: 1 }
+  - match: { transforms.0.id: "simple-local-remote-transform" }
+  - match: { transforms.0.state: "stopped" }
+  - match: { transforms.0.checkpointing.last.checkpoint: 1 }
+
+  # workaround: refresh dest index, to be removed, see gh #TBD
```
Comment (davidkyle) [inline x-pack/qa/multi-cluster-tests-with-security/src/test/resources/rest-api-spec/test/multi_cluster/80_transform.yml]: #TBD -> #51154

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by hendrikmuhs** [comment]
@elasticmachine update branch

**Pull request merged**